tests/integration-tests/utils.py

```diff
@@ -127,6 +128,51 @@ def random_alphanumeric(size=16):
     return "".join(random.choice(string.ascii_lowercase + string.digits) for _ in range(size))
 
 
+def retrieve_metric_data(unique_name, cluster_name, metric_name, period_length_sec, collection_time_min):
+    """Create Boto3 get_metric_data request and output the results"""
+    assert_that(len(unique_name)).is_equal_to(len(metric_name))
+    metric_queries = []
+    for i in range(len(metric_name)):
+        query = {
+            "Id": unique_name[i],
+            "MetricStat": {
+                "Metric": {
+                    "Namespace": "ParallelCluster/Errors/" + cluster_name,
+                    "MetricName": metric_name[i],
+                },
+                "Period": period_length_sec,
+                "Stat": "Sum",
+            },
+        }
+        metric_queries.insert(i, query)
+
+    client = boto3.client("cloudwatch", "us-east-1")
+
+    return client.get_metric_data(
+        MetricDataQueries=metric_queries,
+        StartTime=datetime.now() - timedelta(days=collection_time_min),
+        EndTime=datetime.now() + timedelta(days=collection_time_min),
+        ScanBy="TimestampDescending",
+    )
+
+
+def check_metric_data_query(response, desired_result):
+    """
+    Iterates through get_metric_data query and check for desired results,
+    if desired results = 0 ,check for no change else check if value changed
```

Comment:
Kindly add a sample schema of the `MetricDataResults` in the docstring. Or a link to the CloudWatch API reference for GetMetricData Results.

Reply:
[CloudWatch.Client.get_metric_data](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudwatch.html#CloudWatch.Client.get_metric_data)

Reply:
Decided not to implement this since the idea is to create a troubleshooting link for each group of errors. How I imagine it is that each link displayed in the dashboard will link to a group of more metric specific hyperlinks like in the current troubleshooting guide. Adding a link for each metric creates too much clutter and the current implementation doesn't support that many links just yet

Reply:
Added link to the docstring